src/MacroExpander.js

```diff
@@ -25,18 +32,87 @@ MacroExpander.prototype.nextToken = function() {
         if (!(name.charAt(0) === "\\" && this.macros.hasOwnProperty(name))) {
             return topToken;
         }
+        var tok;
         var expansion = this.macros[name];
         if (typeof expansion === "string") {
+            var numArgs = 0;
+            if (expansion.indexOf("#") !== -1) {
+                var stripped = expansion.replace(/##/g, "");
```

Comment:
Exactly. If I do `\def\setfoo#1{\def\foo##1##2{##1#1##2}}` that's a macro which takes but a single argument, so the string `"\\def\\foo##1##2{##1#1##2}"` should have the same effect.